micro_dl/utils/mp_utils.py

```diff
@@ -221,3 +222,31 @@ def crop_at_indices_save(input_fnames,
         raise e
 
     return tile_meta_df
+
+
+def mp_resize_save(mp_args, workers):
+    """
+    Resize and save images with multiprocessing
+
+    :param dict mp_args: Function keyword arguments
+    :param int workers: max number of workers
+    """
+    with ProcessPoolExecutor(workers) as ex:
+        {ex.submit(resize_and_save, **kwargs): kwargs for kwargs in mp_args}
+
+
```

Comment:
an empirical observation: when using ex.submit followed by sort for tiling tasks, it was slower than using ex.map with no sort. I'm not sure if that'll apply here as well!